poly/src/polynomial/univariate/sparse.rs

```diff
@@ -62,11 +62,28 @@ impl<F: Field> Polynomial<F> for SparsePolynomial<F> {
         if self.is_zero() {
             return F::zero();
         }
+
+        let max_pow_2 = 0usize.leading_zeros() - self.degree().leading_zeros();
+        let mut pows_2 = Vec::<F>::with_capacity(max_pow_2 as usize);
+
+        let mut p = *point;
+        pows_2.push(p);
+        for _ in 1..max_pow_2 {
+            p.square_in_place();
+            pows_2.push(p);
+        }
```

Comment:
I think we can replace it with

```suggestion
		let num_powers = ark_std::log2(self.degree());
        let mut powers_of_2 = Vec::with_capacity(num_powers as usize);

        let mut p = *point;
        pows_2.push(p);
        for _ in 1..=num_powers {
            p.square_in_place();
            powers_of_2.push(p);
        }
```

Reply:
I mean, that's the same as adding 1. and it's not the exact formula as I said...

Reply:
In principle, I am ok with this but I prefer my way as it's more transparent, whereas its not completely clear what log2 does unless you read the impl.